### PR TITLE
[SUP-456] Add startup script support for RStudio

### PR DESCRIPTION
**Pull request by rtitle**
https://broadworkbench.atlassian.net/browse/SUP-456

This PR does the following:
* Renames all occurences of `jupyterUserScript` and `jupyterStartUserScriptUri` to `userScriptUri` and `startUserScriptUri`.
   * Leave old names in API contracts for backwards compatibility (marked deprecated in Swagger)
   * Renames database columns as well
* Bash script changes to support running both user scripts in RStudio containers
* Automation testing for both scripts in GCE/Dataproc * Jupyter/RStudio

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green (you can re-run automation tests with a comment saying `jenkins retest`
- [ ] Run the automation tests multiple times in parallel to weed out instability if applicable via a comment saying `multi-test`
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookGCEClusterMonitoringSpec.scala [hunk 1]
```diff
@@ -1,115 +0,0 @@
...
-import scala.concurrent.duration._
-import scala.util.Try
-
-/**
- * This spec verifies cluster status transitions like pause/resume and cluster PATCH.
- * It is similar in intent to ClusterStatusTransitionsSpec but uses notebooks for validation,
- * so lives in the notebooks sub-package.
- */
-@DoNotDiscover
-class NotebookGCEClusterMonitoringSpec
```
Comment (rtitle) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookGCEClusterMonitoringSpec.scala:22]: This spec only had cases for pause/resume Jupyter and RStudio. That is now covered by `RuntimeGceSpec`, so I removed this file.

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookGCECustomizationSpec.scala [hunk 2]
```diff
@@ -25,43 +22,6 @@ final class NotebookGCECustomizationSpec extends GPAllocFixtureSpec with Paralle
   )
 
   "NotebookGCECustomizationSpec" - {
-
-    "should run a user script" in { billingProject =>
```
Comment (rtitle) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookGCECustomizationSpec.scala:29]: Moved to `RuntimeGceSpec`

File: http/src/main/resources/org/broadinstitute/dsde/workbench/leonardo/liquibase/changesets/20210505_rename_jupyter_user_script_column.xml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog logicalFilePath="leonardo" xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.4.xsd">
+
+    <changeSet logicalFilePath="leonardo" author="rtitle" id="rename_jupyter_user_script_column">
+        <addColumn tableName="CLUSTER">
```
Comment (Qi77Qi) [inline http/src/main/resources/org/broadinstitute/dsde/workbench/leonardo/liquibase/changesets/20210505_rename_jupyter_user_script_column.xml]: think https://docs.liquibase.com/change-types/community/rename-column.html should suffice?

Comment (rtitle) [reply http/src/main/resources/org/broadinstitute/dsde/workbench/leonardo/liquibase/changesets/20210505_rename_jupyter_user_script_column.xml]: Oh yeah, forgot that existed 👍

Comment (rtitle) [reply http/src/main/resources/org/broadinstitute/dsde/workbench/leonardo/liquibase/changesets/20210505_rename_jupyter_user_script_column.xml]: updated

**Comment by rtitle** [comment]
fiab-start failed, jenkins retest

**Comment by rtitle** [comment]
multi-test

**Comment by rtitle** [comment]
some mysql issues, jenkins retest

**Comment by rtitle** [comment]
multi-test

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/RuntimeRoutes.scala [hunk 4]
```diff
@@ -478,58 +481,35 @@ object RuntimeRoutes {
...
-      x.userJupyterExtensionConfig,
-      x.autopauseThreshold,
-      x.defaultClientId,
-      x.clusterImages,
-      x.scopes,
-      x.customClusterEnvironmentVariables,
-      x.diskConfig,
-      x.patchInProgress
+  // can't use Encoder.forProductX because there are 23 fields :(
+  implicit val getRuntimeResponseEncoder: Encoder[GetRuntimeResponse] = Encoder.instance { x =>
```
Comment (Qi77Qi) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/RuntimeRoutes.scala:485]: would be nice to somehow inform users about when we'll remove the deprecated params and have an actual time frame when we plan to do that

File: http/src/main/resources/swagger/api-docs.yaml [hunk 5]
```diff
@@ -2054,24 +2051,33 @@ components:
...
-            inside their jupyter docker. This script runs exactly once when the cluster is first initialized. Logs from
-            this script can be found in the Leo staging bucket for the cluster. Script is run as root
+            Optional GCS or HTTP URI to a bash script the user wishes to run
+            inside their Docker container. This script runs exactly once when the runtime is first initialized. Logs from
+            this script can be found in the Leo staging bucket for the runtime. The script is run as root
             and docker --privileged.
-        jupyterStartUserScriptUri:
+        jupyterUserScriptUri:
+          type: string
+          description: DEPRECATED. Please use userScriptUri instead.
```
Comment (Qi77Qi) [inline http/src/main/resources/swagger/api-docs.yaml]: maybe add the date when it's deprecated? and something like "deprecated fields are subject to be removed after 6 months of deprecation"?

**Review by Qi77Qi — APPROVED** [review]

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/RuntimeRoutes.scala (hunk 4 above)
Comment (rtitle) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/RuntimeRoutes.scala:485]: Yeah.. it's probably just Terra UI and AoU using this field, could log tickets to update each

File: http/src/main/resources/swagger/api-docs.yaml (hunk 5 above)
Comment (rtitle) [reply http/src/main/resources/swagger/api-docs.yaml]: Good idea 👍

**Pull request merged**